cmd/ctr/commands/images/import.go

```diff
@@ -146,33 +145,26 @@ If foobar.tar contains an OCI ref named "latest" and anonymous ref "sha256:deadb
 				opts = append(opts, image.WithNamedPrefix(prefix, overwrite))
 			}
 
-			var platSpec ocispec.Platform
-			// Only when all-platforms not specified, we will check platform value
-			// Implicitly if the platforms is empty, it means all-platforms
+			// Even with --all-platforms, only the default platform layers are unpacked,
+			// for compatibility with --local.
```

Comment:
(Compatibility is out of the scope of ctr, but we cannot ignore that fact that its compatibility is assumed in `kind`: https://github.com/kubernetes-sigs/kind/pull/3828)

Reply:
How was this expected to work? I don't assume ctr remains 100% compatible but it's not clear how you would use `--all-platforms` (Is the expectation that you would have to opt out of the transfer API to use `--all-platforms`?)

Reply:
All platforms for unpack has to be resolved in this case on the transfer side and I think we are missing an ability to express the intent to "unpack all platforms whatever those platforms may be" (which maybe we don't actually want anyway). 

It's pretty hard to get the permutation of options working as expected and especially for cases we may not explicitly want to support, thats one of the reason we don't guarantee anything in ctr because we want that expressiveness without any promises it will always work.

I think just having it keep the original `--local` behavior is fine, we should still consider the expressiveness in the API but I doubt unpack everything is often what a caller wants.